File: hyb-imm-12-2020/lectures/week08/FriYay/prime_numbers.js
// const isPrime = num => {
//   for(let i = 2; i < num; i++)
//     if(num % i === 0) return false;
//   return num > 1;
// }

const isPrime = num => {
    for(let i = 2, s = Math.sqrt(num); i <= s; i++)
        if(num % i === 0) return false; 
    return num > 1;
}

const primeArray = max => {
    arr = [];
    for(let i=2; i < max; i++) {
        if (isPrime(i) === true) {
            arr.push(i);
        }
    }
    return arr;
}

console.log(primeArray(201));

File: hyb-imm-12-2020/lectures/week04/FriYay/prime_numbers.py
max = int(input("Find primes up to what number?: "))

def solution_1(num):
    prime_nums = []
    for x in range(2, num + 1): #range starts at 0
        isPrime = True
        for y in range(2, x):
            if x % y == 0:
                isPrime = False
                break
        if isPrime: 
            prime_nums.append(x)
    return prime_nums


## stop at square root of each number
def solution_2(num):
    prime_nums = []
    for x in range(2, max + 1):
        isPrime = True
        for y in range(2, int(x**0.5)+1):
            if x % y == 0:
                isPrime = False
                break
        if isPrime: 
            prime_nums.append(x)
    return prime_nums


print(solution_2(max))

## solution 3
start = 11
end = 25

for i in range(start,end):
    if i>1:
        for j in range(2,i):
            if(i % j==0):
                break
        else:
            print(i)